lib/app.rb

```diff
@@ -0,0 +1,28 @@
+require 'sinatra/base'
+# require 'byebug'
+# require 'pry'
+require 'tilt/erb'
+require 'data_mapper'
+require 'dm-migrations'
+
+class App < Sinatra::Base
+  # helpers Sinatra::FormHelpers
+  set :views, proc {File.join(root, '..', 'views')}
+  enable :sessions
+  set :session_secret, '121212121212'
+  use Rack::Session::Pool
+  env = ENV['RACK_ENV'] || "development"
+  DataMapper.setup(:default, ENV['DATABASE_URL'] || "postgres://localhost/ohchit_#{env}")
+
+  # DataMapper::Logger.new($stdout, :debug)
+  DataMapper::Model.raise_on_save_failure = true
+  DataMapper.finalize
+  #DataMapper.auto_migrate!
+  DataMapper.auto_upgrade!
+  #byebug
+  get '/' do
+    erb :index
+  end
+  # start the server if ruby file executed directly
+  run! if app_file == $0
+end
```

Comment:
Yes, this is copied over from our bookmarks manager app.  But aren't all of those lines needed, word for word?  Not including, of course, the comment ones, which are ones I added (17, 20, 22, 26), which are all things I used and in bookmarks manager and might want to use here???  I'm confused here because it seems like the copy and paste I did here is not copying and pasting anything that will be unique in any project using Sinatra, Rack, DataMapper, Postgres, and Heroku?